find-lib-in-path.py| Imporved Code| Removed Global

GiorgosXou opened the pull request:
Just made some minor changes +

* Moved gloabal `PREFIX` to the enclosed scope
* "Compressed"\Simplified  the conditional statements under `parse_ld_conf_file`

<sup>Tested with python2 and 3.</sup>

GiorgosXou committed:
find-lib-in-path.py| Imporved Code| Removed Global

albertz commented on bin/find-lib-in-path.py:
```diff
@@ -36,16 +35,13 @@ def get_ld_paths():
+    paths.extend(LDPATH.split(":")) if LDPATH else None 
+    paths.extend(["/lib", "/usr/lib", "/lib64", "/usr/lib64"]) 
+    paths.extend([PREFIX, PREFIX + "/lib", PREFIX + "/usr/lib", PREFIX + "/lib64", PREFIX + "/usr/lib64"]) if PREFIX else None
+    paths.extend(parse_ld_conf_file("/etc/ld.so.conf")) if os.path.exists("/etc/ld.so.conf") else print('WARNING: file "/etc/ld.so.conf" not found.')
```
This is unpythonic code style. I prefer more standard code style (which is easier to read for most people, as it is more standard).

GiorgosXou committed:
Improved Code | Removed Global & Extend

GiorgosXou commented:
That's fine, I changed things back to how they were, except I:

* Moved gloabal `PREFIX` to the enclosed scope
* And replaced `paths.extend(["/lib", "/usr/lib", "/lib64", "/usr/lib64"])` with `path = ["/lib", "/usr/lib", "/lib64", "/usr/lib64"]`

albertz commented on bin/find-lib-in-path.py:
```diff
@@ -36,16 +35,17 @@ def get_ld_paths():
+    if LDPATH: 
+        paths.extend(LDPATH.split(":"))
+    if PREFIX:
+        paths.extend([PREFIX, PREFIX + "/lib", PREFIX + "/usr/lib", PREFIX + "/lib64", PREFIX + "/usr/lib64"])
```
I just notice that you changed the order. The order is relevant though as it should reflect the actual search paths for executables.

(I also wonder now, which executables actually look at `PREFIX` for libraries?)

GiorgosXou replied:
You got me there, I wasn't  really sure if executables actually also look at the exact `PREFIX` path for libraries and so i've added it just to be sure, (sorry for that). [of course all the other `PREFIX +`  have to be there]

GiorgosXou replied:
> it should reflect the actual search paths for executables.

in what way it should reflect the actual search paths for executables? i don't get it :thinking:

albertz replied:
Executables when they load libraries (e.g. `libc.so`) will search through these paths in a well defined order.

GiorgosXou replied:
So should i remove `paths = ["/lib", "/usr/lib", "/lib64", "/usr/lib64"]` and re-add line `paths.extend(['/lib'...]`  ?

GiorgosXou replied:
You mean `LDPATH`s should be checked \\`extend`ed  before `"/lib", "/usr/lib",  "/lib64", "/usr/lib64"` and so on ..?

GiorgosXou replied:
not sure how the order should be...

albertz replied:
Yes right. So imagine you have `libc.so` in multiple of those directories, and you start any executable which tries to load `libc.so`. Then the order matters. It will load the `libc.so` from the first directory. The order is well defined. And this is basically what I wrote in the comment in the beginning of the function:

> To be very correct, see man-page of ld.so.
> And here: http://unix.stackexchange.com/questions/354295/what-is-the-default-value-of-ld-library-path/354296
> Short version, not specific to an executable, in this order:
> - LD_LIBRARY_PATH
> - /etc/ld.so.cache (instead we will parse /etc/ld.so.conf)
> - /lib, /usr/lib (or maybe /lib64, /usr/lib64)

GiorgosXou replied:
Ok, order is fixed now :+1:

albertz commented on bin/find-lib-in-path.py:
```diff
@@ -36,16 +35,17 @@ def get_ld_paths():
+        paths.extend(LDPATH.split(":"))
+    if PREFIX:
+        paths.extend([PREFIX, PREFIX + "/lib", PREFIX + "/usr/lib", PREFIX + "/lib64", PREFIX + "/usr/lib64"])
     if os.path.exists("/etc/ld.so.conf"):
```
Btw, maybe you also want to check the same for `f"{PREFIX}/etc/ld.so.conf"`?

GiorgosXou replied:
Indeed, thanks for the catch, i completely ignored it

GiorgosXou replied:
* Removed PREFIX
* Added check for `PREFIX +  "/etc/ld.so.conf"`

GiorgosXou committed:
Update find-lib-in-path.py

GiorgosXou commented:
*(it's late here, I'm going to sleep now :sleeping: )*
![image](https://user-images.githubusercontent.com/11987271/141185656-c6c74545-6694-4f88-85b4-d31ab4870c03.png)
<sup>^ you can edit whatever you want thought</sup>

GiorgosXou committed:
Update find-lib-in-path.py

GiorgosXou commented:
I think now it's fine

albertz commented:
Looks good.
